docs/architecture/node-high-level.md

```diff
@@ -1 +1,117 @@
 # Node (high level)
+
+Mysterium network will act as a decentralized marketplace between providers and consumers participating in building and maintaining this infrastructure.
+
+![image](draw.io/node-high-level.png)
+
+## Service provider
+
+Any Mysterium network node can become a Service provider and provide VPN services using his own infrastructure. To become a service provider, the following requirements should be done:
+* Create personal user identity;
+* Register user identity;
+* Create a service proposal;
+* Register service proposal in the Discovery service.
+
+All Mysterium network nodes should have a valid identity to interact with other systems. `myst` binary provides a way to create a required identity.
+
+Once identity created, the user should finish the identity registration process to be able to provide services in the Mysterium network.
+
+Registered identity is used as one of the components for creating a service proposal, that acts as a complete description of the service that Service provider going to provide.
+
+The created proposal should be registered in the service discovery component to allow other network participants to search through all available services and consume it.
+
+At that moment the service provider starts to wait for an incoming request. It handles all valid requests and establishes a new VPN session for the user.
+
+The service provider sends to the consumer all required information for establishing VPN session.
+
+## Service consumer
+
+The same `myst` binary could be used as a service consumer. The service consumer should have a similar registered identity as a service provider to be able to communicate with it.
+
+Once consumer identity created and registered in the network, a user can request discovery service for the list of available service proposals. The results of this request can be filtered through the discovery API.
+
+Service consumer chooses the service proposal that fit his requirements and starts communication dialog to negotiate session creation.
+
+Service consumer gets from the provider information required for establishing VPN session. It initializes the VPN session creation process.
+
+When the VPN session established, the service consumer can consume provided services as he needs. And service consumer can close the session any time he wants.
```

Comment:
When the VPN session is established, the service consumer can consume the provided services as he needs. The service consumer can close the session at any time if he wishes to do so.